terminus-ui/tooltip/src/tooltip.component.spec.ts

```diff
@@ -1,46 +1,139 @@
+import { showTooltip } from '../testing/src/test-helpers';
 
 
 describe(`TsTooltipComponent`, function() {
-  let component: TsTooltipComponent;
 
-  beforeEach(() => {
-    component = new TsTooltipComponent();
-  });
+  describe(`Defaults`, () => {
+    let component: testComponents.Basic;
+    let fixture: ComponentFixture<testComponents.Basic>;
+    let tooltipEl: HTMLElement;
+    let tooltipComponent: TsTooltipComponent;
 
+    beforeEach(() => {
+      fixture = createComponent(testComponents.Basic, [], [TsTooltipModule]);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
+      tooltipComponent = component.tooltipComponent;
 
-  it(`should exist`, () => {
-    expect(component).toBeTruthy();
-  });
+      tooltipEl = fixture.debugElement.query(By.css('.c-tooltip')).nativeElement as HTMLElement;
+    });
 
-  describe(`hasUnderline`, () => {
-    test(`should set and retrieve`, () => {
-      component.hasUnderline = true;
-      expect(component.hasUnderline).toEqual(true);
+    test(`should exist`, () => {
+      expect(tooltipComponent).toBeTruthy();
+      expect(tooltipEl).toBeTruthy();
+    });
+
+    test(`should set position to below`, () => {
+      expect(tooltipComponent.tooltipPosition).toEqual('below');
+      expect(tooltipEl.attributes.getNamedItem('ng-reflect-position').value).toEqual('below');
+    });
+
+    test('should set underline to false', () => {
+      expect(tooltipComponent.hasUnderline).toEqual(false);
+      expect(tooltipEl.classList).not.toContain('c-tooltip--underline');
     });
+
   });
 
-  describe(`tooltipPosition`, () => {
+  describe('Customized', () => {
+    let component: testComponents.TestHostComponent;
+    let fixture: ComponentFixture<testComponents.TestHostComponent>;
+    let tooltipEl: HTMLElement;
+    let tooltipComponent: TsTooltipComponent;
 
-    test(`should set below if no position is declared`, () => {
-      expect(component.tooltipPosition).toEqual('below');
+    beforeEach(() => {
+      fixture = createComponent(testComponents.TestHostComponent, [], [TsTooltipModule]);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
+      tooltipComponent = component.tooltipComponent;
+      tooltipEl = fixture.debugElement.query(By.css('.c-tooltip')).nativeElement as HTMLElement;
+    });
 
+    test(`should exist`, () => {
+      expect(tooltipComponent).toBeTruthy();
+      expect(tooltipEl).toBeTruthy();
     });
 
-    test(`should set correct position`, () => {
-      component.tooltipPosition = 'above';
-      expect(component.tooltipPosition).toEqual('above');
+    describe(`tooltipPosition`, () => {
+      test(`should set and retrieve above`, () => {
+        component.tooltipPosition = 'above';
+        fixture.detectChanges();
 
-    });
+        expect(tooltipComponent.tooltipPosition).toEqual('above');
+        expect(tooltipEl.attributes.getNamedItem('ng-reflect-position').value).toEqual('above');
```

Comment:
I don't think we should be testing `ng-*` data attributes directly. We have no control over those values. We should probably be testing for positions or classes. (same comment below)